### PR TITLE
Dillon odle newsletter

### PR DESCRIPTION
**Pull request by DillonOdle**
### Client Changes

- Added the form to submit the newsletter subscription and added it to the footer links

### Server Changes

- Installed npm package `nodemailer`
- Created a model for the MongoDB collection for newsletter subscribers.
- Created the routes to sign up to the newsletter (/newsletter/signup) and to verify the jwt from the verification email (/newsletter/verify/:token).

### CODE REVIEW

File: client/src/components/FooterLinks/FooterLinks.tsx [hunk 1]
```diff
@@ -2,6 +2,9 @@ import { Text, Container, Image, ActionIcon, Group, rem } from '@mantine/core';
 import { IconBrandInstagram, IconBrandFacebook } from '@tabler/icons-react';
 import logoImage from '../../images/logo.png';
 import classes from './FooterLinks.module.css';
+import React, { useState, ChangeEvent, FormEvent } from 'react';
```
Comment (steventran619) [inline client/src/components/FooterLinks/FooterLinks.tsx]: Remove `React`
_src/components/FooterLinks/FooterLinks.tsx:5:8 - error TS6133: 'React' is declared but its value is never read._

File: client/src/components/FooterLinks/FooterLinks.tsx [hunk 2]
```diff
@@ -53,6 +66,96 @@ export function FooterLinks() {
...
+    });
+  };
+
+  // Add the form as another group within the groups container
+  groups.push(
+    <div className={classes.wrapper} key="newsletter">
+      <form onSubmit={handleNewsletterSubmit} className={classes.form}>
+        <Text className={classes.title}>Subscribe to our newsletter</Text>
+        <div className={classes.formField}>
+          <label htmlFor="email" className={classes.label}>Email</label>
```
Comment (steventran619) [inline client/src/components/FooterLinks/FooterLinks.tsx]: I'd probably remove the labels and just keep the fields as "Email", "First Name", "Last Name" to minimize the footer.

Comment (steventran619) [reply client/src/components/FooterLinks/FooterLinks.tsx]: Some examples
image
image
image
image

Comment (DillonOdle) [reply client/src/components/FooterLinks/FooterLinks.tsx]: Good idea!

**Review by steventran619 — APPROVED** [review]
Looks good; approved!

image

**Pull request merged**